src/GPUActivityAgent.cpp

```diff
@@ -37,7 +37,11 @@ namespace geopm
         , M_WAIT_SEC(0.020) // 20ms wait default
         , M_POLICY_PHI_DEFAULT(0.5)
         , M_GPU_ACTIVITY_CUTOFF(0.05)
-        , M_NUM_GPU(m_platform_topo.num_domain(GEOPM_DOMAIN_GPU))
+        , M_NUM_GPU(m_platform_topo.num_domain(
+                    GEOPM_DOMAIN_GPU))
+        , M_NUM_GPU_CHIP(m_platform_topo.num_domain(
+                    GEOPM_DOMAIN_GPU_CHIP))
+        , M_NUM_CHIP_PER_GPU(M_NUM_GPU_CHIP/M_NUM_GPU)
```

Comment:
Currently in LevelZero we check that the number of sub-devices is evenly divisible by the number of devices:
https://github.com/geopm/geopm/blob/dev/service/src/LevelZero.cpp#L143

However it doesn't fully address the issue that in several locations (including this agent) we make the assumption that ever device will have the same number of sub-devices.  
This may be worth enforcing in GPUTopo, LevlZeroDevicePool, or LevelZero.cpp